lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQuerySqlProvider.java

```diff
@@ -0,0 +1,425 @@
+  private static final String OPENING_WHERE_TAG = "<where>";
+  private static final String CLOSING_WHERE_TAG = "</where>";
+  private static final String WILDCARD_LIKE_STATEMENT =
+      "<if test='wildcardSearchValueLike != null and wildcardSearchFieldIn != null'>AND ("
+          + "<foreach item='item' collection='wildcardSearchFieldIn' separator=' OR '>"
+          + "UPPER(t.${item}) "
+          + "LIKE #{wildcardSearchValueLike}"
+          + "</foreach>)"
+          + "</if> ";
+
+  private TaskQuerySqlProvider() {}
+
+  public static String queryTaskSummaries() {
+    return OPENING_SCRIPT_TAG
+        + "SELECT <if test=\"useDistinctKeyword\">DISTINCT</if> "
+        + "t.ID, t.EXTERNAL_ID, t.CREATED, t.CLAIMED, t.COMPLETED, t.MODIFIED, t.PLANNED, t.DUE, "
+        + "t.NAME, t.CREATOR, t.DESCRIPTION, t.NOTE, t.PRIORITY, t.STATE, t.CLASSIFICATION_KEY, "
+        + "t.CLASSIFICATION_CATEGORY, t.CLASSIFICATION_ID, t.WORKBASKET_ID, t.DOMAIN, "
+        + "t.WORKBASKET_KEY, t.BUSINESS_PROCESS_ID, t.PARENT_BUSINESS_PROCESS_ID, t.OWNER, "
+        + "t.POR_COMPANY, t.POR_SYSTEM, t.POR_INSTANCE, t.POR_TYPE, t.POR_VALUE, t.IS_READ, "
+        + "t.IS_TRANSFERRED, t.CUSTOM_1, t.CUSTOM_2, t.CUSTOM_3, t.CUSTOM_4, t.CUSTOM_5, "
+        + "t.CUSTOM_6, t.CUSTOM_7, t.CUSTOM_8, t.CUSTOM_9, t.CUSTOM_10, t.CUSTOM_11, t.CUSTOM_12, "
+        + "t.CUSTOM_13, t.CUSTOM_14, t.CUSTOM_15, t.CUSTOM_16"
+        + "<if test=\"addAttachmentColumnsToSelectClauseForOrdering\">"
+        + ", a.CLASSIFICATION_ID, a.CLASSIFICATION_KEY, a.CHANNEL, a.REF_VALUE, a.RECEIVED"
+        + "</if>"
+        + "<if test=\"addClassificationNameToSelectClauseForOrdering\">, c.NAME </if>"
+        + "<if test=\"addAttachmentClassificationNameToSelectClauseForOrdering\">, ac.NAME </if>"
+        + "<if test=\"addWorkbasketNameToSelectClauseForOrdering\">, w.NAME </if>"
+        + "FROM TASK t "
+        + "<if test=\"joinWithAttachments\">"
+        + "LEFT JOIN ATTACHMENT AS a ON t.ID = a.TASK_ID "
+        + "</if>"
+        + "<if test=\"joinWithClassifications\">"
+        + "LEFT JOIN CLASSIFICATION AS c ON t.CLASSIFICATION_ID = c.ID "
+        + "</if>"
+        + "<if test=\"joinWithAttachmentClassifications\">"
+        + "LEFT JOIN CLASSIFICATION AS ac ON a.CLASSIFICATION_ID = ac.ID "
+        + "</if>"
+        + "<if test=\"joinWithWorkbaskets\">"
+        + "LEFT JOIN WORKBASKET AS w ON t.WORKBASKET_ID = w.ID "
+        + "</if>"
+        + OPENING_WHERE_TAG
+        + "<if test='accessIdIn != null'> "
+        + "AND t.WORKBASKET_ID IN ("
+        + "SELECT WID from ("
+        + "SELECT WORKBASKET_ID "
+        + "as WID, MAX(PERM_READ::int) "
+        + "as MAX_READ FROM WORKBASKET_ACCESS_LIST "
+        + "AS s where ACCESS_ID IN "
+        + "(<foreach item='item' collection='accessIdIn' separator=',' >#{item}</foreach>) "
+        + "group by WORKBASKET_ID) "
+        + "AS f where max_read = 1) "
+        + "</if> "
```

Comment:
let's extract this to a method. A `checkForAuthorization` is more readable than this :)